content/blog/r-frictionless-validation.md

```diff
@@ -0,0 +1,85 @@
+---
+title: "Frictionless Data Package Validation in R"
+date: 2023-03-30
+author: Rob Ashton
+tags:
+ - R
+ - frictionless
+ - data
+ - validation
+---
+
+Our work ofen involves researcher-written models that we want to make available to non-technical users. To ensure that both the model will run and that it will produce a good fit, validating users data and returning actionable error messages is essential. For the [UNAIDS Naomi project](/projects/naomi) we collaborate closely with a team that manages data storage for users. They use Frictionless Data Packages for describing the resources that need to be present and validating them. When users upload their data to our application, we run separate validation code written in R. It would be ideal to validate with a common code base to avoid duplicated work and ensure consistency both in messaging.
+
+One way we could support this is by using the [Frictionless Data Package](https://frictionlessdata.io/) schema developed by our partners to validate the data from R. This blog post is an exploration of the capabilities of Frictionless for validation from R.
+
+User upload files of different types including tabular data (as CSVs), geojson and more exotic types such as PJNZ (which really is a zip with files containing hybrid xml & csv data). The checks we run on these range from very simple assertions to more complicated and bespoke ones. Simple assertions include checking that data has certain column, the values in the columns are from a set of possible values, and that a column is numeric, positive numeric or matches a regex pattern. More complicated checks include ensuring that geojson has a single parent region, that every feature in geojson has a property with a certain name, that there is a single row for each combination of some category columns. We also sometimes want to validate across two files for example to check that the values in a CSV column match one of the properties from a geojson file.
+ 
+In this post we will look at two R packages for working with Frictionless Data Packages: `frictionless` and `datapackage.r`. Both are developed by Frictionless Data organisation. 
```

Comment:
Either at the beginning of the post or the end you need your conclusion: having investigated these packages they don't fit what we need and we're exploring other ways of reducing duplication, etc. Could do it up front and say "below is a list of issues that we found" or below as a conclusion section and perhaps outline possible ways forward